azext_iot/central/providers/device_provider.py

```diff
@@ -190,6 +202,108 @@ def delete_device(
 
         return result
 
+    def list_relationships(
+        self,
+        device_id,
+        rel_name=None,
+        central_dns_suffix=CENTRAL_ENDPOINT,
+    ) -> List[RelationshipV1_1_preview]:
+        relationships = central_services.device.list_relationships(
+            self._cmd,
+            app_id=self._app_id,
+            device_id=device_id,
+            token=self._token,
+            api_version=self._api_version,
+            central_dns_suffix=central_dns_suffix,
+        )
+
+        if relationships is None:
+            raise ResourceNotFoundError(
+                "No relationships found for device with id: '{}'.".format(device_id)
+            )
+
+        if rel_name:
+            relationships = [rel for rel in relationships if rel.name == rel_name]
+
+        return relationships
+
+    def add_relationship(
+        self,
+        device_id,
+        target_id,
+        rel_id,
+        rel_name,
+        central_dns_suffix=CENTRAL_ENDPOINT,
+    ) -> dict:
+        relationship = central_services.device.create_relationship(
+            self._cmd,
+            app_id=self._app_id,
+            device_id=device_id,
+            rel_id=rel_id,
+            rel_name=rel_name,
+            target_id=target_id,
+            token=self._token,
+            api_version=self._api_version,
+            central_dns_suffix=central_dns_suffix,
+        )
+
+        if not relationship:
```

Comment:
? if you can't add a relationship, the error should say so. Does the service not return errors with the specific reason? (for example the device id was not found, the target id was not found, etc)

Reply:
backend API does not return specific reason. Just a "resource not found" that can be related to either missing source or target.